napari/layers/shapes/shape_models/shape.py

```diff
@@ -208,39 +213,44 @@ def edge_width(self, edge_width):
         self._edge_width = edge_width
 
     @property
-    def edge_color(self):
-        """Color, ColorArray: color of edges
-        """
+    def edge_color(self) -> np.ndarray:
+        """RGBA array of the color of the shape's edges."""
         return self._edge_color
 
     @edge_color.setter
-    def edge_color(self, edge_color):
-        self._edge_color = Color(edge_color)
-        if type(edge_color) is str:
-            self._edge_color_name = edge_color
-        else:
-            rgb = tuple([int(255 * x) for x in self._edge_color.rgba[:3]])
-            self._edge_color_name = '#%02x%02x%02x' % rgb
+    def edge_color(self, edge_color: np.ndarray):
+        """Change the edge color of the shape.
+
+        This method assumes that the incoming data passed through
+        transformation (``transform_color_with_defaults``) and
+        normalization (``normalize_and_broadcast_colors``) which
+        means it's already an 1x4 numpy array.
+        """
+        self._edge_color = edge_color
+        hexval = rgb_to_hex(self._edge_color)
+        self._edge_color_name = hex_to_name.get(hexval, hexval)
```

Comment:
Since this class will not be instatiated by a user I allowed it to set the internal `_edge_color` attribute without any further checks. All checks happen upstream in the child classes.